src/main/java/ro/unibuc/triplea/domain/games/steam/service/SteamGameService.java

```diff
@@ -39,16 +38,14 @@ public Optional<SteamGameResponse> getGameByIdentifier(String identifier) {
             } else {
                 throw new SteamGameNotFoundException("Steam game with identifier " + identifier + " not found");
             }
-        } else if (IdentifierUtil.isValidString(identifier)) {
+        } else {
```

Comment:
Refactor this method. Instead of coding replicated code for if and else branch, extract functionality for both of them, based on conditions.

Also, a clean code is checking for errors first and then treat positive case accordingly.

Example: https://medium.com/@brooknovak/why-you-should-reduce-nesting-blocks-in-your-code-with-practical-refactoring-tips-11c122735559

Name of that is also 'nesting hell'.